mods/flowers/init.lua

```diff
@@ -145,26 +145,36 @@ function flowers.flower_spread(pos, node)
 
 	local soils = minetest.find_nodes_in_area_under_air(
 		pos0, pos1, "group:soil")
-	if #soils > 0 then
-		local seedling = soils[math.random(#soils)]
-		local seedling_above =
-			{x = seedling.x, y = seedling.y + 1, z = seedling.z}
-		light = minetest.get_node_light(seedling_above)
-		if not light or light < 13 or
-				-- Desert sand is in the soil group
-				minetest.get_node(seedling).name == "default:desert_sand" then
-			return
+	local num_soils = #soils
+	if num_soils > 0 then
+		-- Get 2 random soil nodes from area
+		local soil_pos = {}
+		soil_pos[1] = soils[math.random(num_soils)]
+		soil_pos[2] = soils[math.random(num_soils)]
+		while soil_pos[2] == soil_pos[1] do
+			soil_pos[2] = soils[math.random(num_soils)]
```

Comment:
Possible infinite loop if there's only one soil.

Reply:
Well spotted.

Reply:
Avoided this by simplifying and not worrying about duplicated positions.

Reply:
Duplicated positions are unlikely with the typical large areas of soils, avoiding them requires lots of additional code.